### PR TITLE
refactor: Make constants enums top level and `PascalCase`

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Review by ImRodry — CHANGES_REQUESTED** [review]
Two questions: why are these constants being moved to their own files if they're still constants? And why are they PascalCased if they're immutable constants, which are [usually](https://www.robinwieruch.de/javascript-naming-conventions/) UPPERCASED in JavaScript?

**Comment by suneettipirneni** [comment]
> Two questions: why are these constants being moved to their own files if they're still constants? And why are they PascalCased if they're immutable constants, which are [usually](https://www.robinwieruch.de/javascript-naming-conventions/) UPPERCASED in JavaScript?

Just to be clear, enums themselves are namespaces of immutable constant values, and we use PascalCase for those. Whenever we migrate to typescript these constant objects will most definitely be converted to ts enums.

**Comment by ImRodry** [comment]
The biggest benefit I see and the reason why they were there to begin with was organization, so that we can have all enums in one single place. They’re still all immutable constants so I think they should stay UPPERCASED like they were before

**Comment by kyranet** [comment]
Namespaces and enums are and still should be PascalCased, including those that aren't but look like it, as @suneettipirneni said.

This change also brings consistency in the project, so yeah, that's a final say.

**Review by kyranet — APPROVED** [review]
Only thing I'd change would be changing `module.exports = createEnum(...);` to `exports.EnumName = createEnum(...);` so we have 1 thing less to do when we rewrite to TypeScript, but this is fine too.

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

File: packages/discord.js/src/client/websocket/handlers/THREAD_UPDATE.js [hunk 1]
```diff
@@ -11,6 +11,6 @@ module.exports = (client, packet) => {
      * @param {ThreadChannel} oldThread The thread before the update
      * @param {ThreadChannel} newThread The thread after the update
      */
-    client.emit(Events.THREAD_UPDATE, old, updated);
+    client.emit(Events.ThreadCreate, old, updated);
```
Comment (vitorlops) [inline packages/discord.js/src/client/websocket/handlers/THREAD_UPDATE.js:14]:
```suggestion
    client.emit(Events.ThreadUpdate, old, updated);
```
Isn't it supposed to be `ThreadUpdate`?

File: packages/discord.js/src/structures/InteractionCollector.js [hunk 2]
```diff
@@ -92,31 +92,31 @@ class InteractionCollector extends Collector {
 
     if (this.messageId) {
       this._handleMessageDeletion = this._handleMessageDeletion.bind(this);
-      this.client.on(Events.MESSAGE_DELETE, this._handleMessageDeletion);
-      this.client.on(Events.MESSAGE_BULK_DELETE, bulkDeleteListener);
+      this.client.on(Events.MessageDelete, this._handleMessageDeletion);
+      this.client.on(Events.MessageDelete, bulkDeleteListener);
```
Comment (vitorlops) [inline packages/discord.js/src/structures/InteractionCollector.js:96]:
```suggestion
      this.client.on(Events.MessageBulkDelete, bulkDeleteListener);
```

**Comment by iCrawl** [comment]
This needs a rebase.

File: packages/discord.js/src/client/websocket/WebSocketShard.js [hunk 3]
```diff
@@ -4,8 +4,12 @@ const EventEmitter = require('node:events');
 const { setTimeout, setInterval, clearTimeout, clearInterval } = require('node:timers');
 const { GatewayIntentBits } = require('discord-api-types/v9');
 const WebSocket = require('../../WebSocket');
-const { Status, Events, ShardEvents, Opcodes, WSEvents } = require('../../util/Constants');
+const { WSEvents } = require('../../util/Constants');
+const Events = require('../../util/Events');
 const IntentsBitField = require('../../util/IntentsBitField');
+const Opcodes = require('../../util/Opcodes');
```
Comment (vladfrangu) [inline packages/discord.js/src/client/websocket/WebSocketShard.js:10]: Isn't this an enum in -types?

File: packages/discord.js/src/managers/RoleManager.js [hunk 4]
```diff
@@ -128,7 +128,7 @@ class RoleManager extends CachedManager {
    * // Create a new role with data and a reason
    * guild.roles.create({
    *   name: 'Super Cool Blue People',
-   *   color: 'BLUE',
+   *   color: 'Blue',
```
Comment (vladfrangu) [inline packages/discord.js/src/managers/RoleManager.js:131]: 🤔

File: packages/discord.js/typings/index.d.ts [hunk 5]
```diff
@@ -3657,140 +3652,141 @@ export interface CommandInteractionResolvedData<Cached extends CacheType = Cache
...
-}
-
-export interface ConstantsStatus {
-  READY: 0;
-  CONNECTING: 1;
-  RECONNECTING: 2;
-  IDLE: 3;
-  NEARLY: 4;
-  DISCONNECTED: 5;
+export enum Colors {
```
Comment (vladfrangu) [inline packages/discord.js/typings/index.d.ts:3655]: This isn't an actual enum but an object

File: packages/discord.js/typings/index.d.ts [hunk 6]
```diff
@@ -3657,140 +3652,141 @@ export interface CommandInteractionResolvedData<Cached extends CacheType = Cache
...
+  DarkerGrey = 0x7f8c8d,
+  LightGrey = 0xbcc0c0,
+  DarkNavy = 0x2c3e50,
+  Blurple = 0x5865f2,
+  Greyple = 0x99aab5,
+  DarkButNotBlack = 0x2c2f33,
+  NotQuiteBlack = 0x23272a,
+}
+
+export enum Events {
```
Comment (vladfrangu) [inline packages/discord.js/typings/index.d.ts:3688]: Ditto

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**